compositor_pipeline/src/queue/video_queue_thread.rs

```diff
@@ -71,7 +72,14 @@ impl VideoQueueThread {
         for input_id in frames_batch.frames.keys() {
             internal_queue.call_input_listeners(input_id)
         }
-        self.sender.send(frames_batch).unwrap();
+        let send_deadline = self.opts.clock_start.add(frames_batch.pts);
+        if self
+            .sender
+            .send_deadline(frames_batch, send_deadline)
+            .is_err()
+        {
+            warn!("Dropping video frame on queue output.")
+        }
```

Comment:
This deadline looks wrong to me. This exact timestamp looks like the time when output frames rendering should start. What happens if rendering takes some time? Won't this stop sending any frames to output?

Reply:
discussed offline